src/server/req_delete.c

```diff
@@ -963,23 +975,6 @@ req_deletejob2(struct batch_request *preq, job *pjob)
 			return 0;
 		}
 
-	} else if ((check_job_substate(pjob, JOB_SUBSTATE_PRERUN)) && (forcedel == 0)) {
```

Comment:
what about transit? we don't need to defer for that either right?

Reply:
We rely on the outcome of the movejob to reply back to the client. If the move is successful, we return an unknown job ID so the client will go ahead with locating job and then deleting it. If the move has failed, then we will delete the job and return a success.

Reply:
that doesn't answer my question though, do we need to defer deljoblist for that? can't we put it in AVL tree and move on? can we update the job object when the move completes, just like we do for prerun state jobs?

Reply:
For prerun we send the Termjob signal to the mom and moves on with deletion.
We cannot do that with the moved jobs as we rely on the outcome of the move to respond to the client.
We can of course handle it with special logic in post_movejob or by having a special list which we can check them later. But that effort seem doesn't worth it as moving job is not something we will encounter frequently.